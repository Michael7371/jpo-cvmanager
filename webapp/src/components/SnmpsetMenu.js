import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import {
  selectChangeSuccess,
  selectErrorState,
  selectDestIp,
  selectSnmpMsgType,
  selectSnmpFilterMsg,
  selectSnmpFilterErr,

  // Actions
  submitSnmpSet,
  filterSnmp,
  setDestIp,
  setMsgType,
} from '../generalSlices/configSlice'

import { selectRsuIpv4, selectRsuManufacturer } from '../generalSlices/rsuSlice'

import './css/SnmpwalkMenu.css'

const SnmpsetMenu = () => {
  const dispatch = useDispatch()

  const changeSuccess = useSelector(selectChangeSuccess)
  const errorState = useSelector(selectErrorState)
  const destIp = useSelector(selectDestIp)
  const snmpMsgType = useSelector(selectSnmpMsgType)
  const snmpFilterMsg = useSelector(selectSnmpFilterMsg)
  const snmpFilterErr = useSelector(selectSnmpFilterErr)

  const rsuIp = useSelector(selectRsuIpv4)
  const rsuManufacturer = useSelector(selectRsuManufacturer)

  return (
    <div id="snmpdiv">
      <h2 id="snmpheader">Add Message Forwarding</h2>

      <form id="snmpform">
        <label id="snmplabel">
          <strong>Destination IP:</strong>
          <input id="snmpinput" type="text" value={destIp} onChange={(e) => dispatch(setDestIp(e.target.value))} />
        </label>
        <label id="snmplabel">
          <strong>Message Type:</strong>
          <select id="snmpdropdown" value={snmpMsgType} onChange={(e) => dispatch(setMsgType(e.target.value))}>
            <option value="bsm">BSM</option>
            <option value="spat">SPaT</option>
            <option value="map">MAP</option>
            <option value="srm">SRM</option>
            <option value="ssm">SSM</option>
          </select>
        </label>
      </form>

      <button id="refreshbtn" onClick={() => dispatch(submitSnmpSet([rsuIp]))}>
        Submit Config
      </button>

      {changeSuccess ? (
        <div>
          <p id="successtext">Successful write to RSU</p>
          <p id="infotext">Only message type and index is required for delete</p>
        </div>
      ) : (
        <p id="infotext">Only message type and index is required for delete</p>
      )}
      {errorState !== '' ? <p id="warningtext">{errorState}</p> : <div />}

      {rsuManufacturer === 'Yunex' ? (
        <div>
          <p id="snmpfiltertext" marginTop="40px">
            Yunex RSUs use different SNMP tables for message TX and RX forwarding. <br /> BSM and SSM are on the RX
            table. MAP, SPaT and SRM are on the TX table. <br /> Start over from the 1 index for each table.
          </p>
        </div>
      ) : (
        <div />
      )}

      {rsuManufacturer === 'Commsignia' ? (
        <div>
          <p id="snmpfiltertext" marginTop="40px">
            If you are configuring SPaT or MAP forwarding, apply the TX message <br /> filter after your configuration
            has been applied
          </p>
          <button id="refreshbtn" onClick={() => dispatch(filterSnmp([rsuIp]))}>
            Apply TX Filter
          </button>
          {snmpFilterMsg !== '' ? (
            <div>
              {snmpFilterErr === true ? (
                <p id="warningtext">{snmpFilterMsg}</p>
              ) : (
                <p id="successtext">{snmpFilterMsg}</p>
              )}
            </div>
          ) : (
            <div />
          )}
        </div>
      ) : (
        <div />
      )}
    </div>
  )
}

export default SnmpsetMenu
