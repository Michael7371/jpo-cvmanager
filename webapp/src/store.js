import { configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import rsuUpdateMenuReducer from './features/rsuUpdateMenu/rsuUpdateMenuSlice'

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      rsuUpdateMenu: rsuUpdateMenuReducer,
    },
    preloadedState,
  })
}
