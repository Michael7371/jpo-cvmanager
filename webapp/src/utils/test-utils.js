/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
