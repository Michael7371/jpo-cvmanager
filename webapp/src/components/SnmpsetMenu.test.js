import React from 'react'
import { render } from '@testing-library/react'
import SnmpsetMenu from './SnmpsetMenu'
import { Provider } from 'react-redux'
import { setupStore } from '../store'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(
    <Provider store={setupStore({})}>
      <SnmpsetMenu />
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
