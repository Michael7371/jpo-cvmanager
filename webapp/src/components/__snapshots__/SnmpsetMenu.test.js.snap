// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="snmpdiv"
  >
    <h2
      id="snmpheader"
    >
      Add Message Forwarding
    </h2>
    <form
      id="snmpform"
    >
      <label
        id="snmplabel"
      >
        <strong>
          Destination IP:
        </strong>
        <input
          id="snmpinput"
          type="text"
          value=""
        />
      </label>
      <label
        id="snmplabel"
      >
        <strong>
          Message Type:
        </strong>
        <select
          id="snmpdropdown"
        >
          <option
            value="bsm"
          >
            BSM
          </option>
          <option
            value="spat"
          >
            SPaT
          </option>
          <option
            value="map"
          >
            MAP
          </option>
          <option
            value="srm"
          >
            SRM
          </option>
          <option
            value="ssm"
          >
            SSM
          </option>
        </select>
      </label>
    </form>
    <button
      id="refreshbtn"
    >
      Submit Config
    </button>
    <p
      id="infotext"
    >
      Only message type and index is required for delete
    </p>
    <div />
    <div />
    <div />
  </div>
</div>
`;
